### PR TITLE
Fixes on jwt_svid parse and validate.

### PR DESCRIPTION
**Pull request by glaucimar**
- fixing the missing sub claim in token, adidng test for it, adding typing for test function

### CODE REVIEW

File: src/pyspiffe/svid/jwt_svid.py [hunk 1]
```diff
@@ -132,7 +132,7 @@ def parse_and_validate(
                     'verify_exp': True,
                 },
             )
-            spiffe_ID = SpiffeId.parse(claims['sub'])
+            spiffe_ID = SpiffeId.parse(claims.get('sub', None))
```
Comment (Andres-GC) [inline src/pyspiffe/svid/jwt_svid.py:135]: nit: [`get()`](https://docs.python.org/3/library/stdtypes.html#dict.get) default value is `None` so it is not necessary to add it here.

File: test/svid/jwtsvid/test_jwt_svid.py [hunk 2]
```diff
@@ -231,6 +231,29 @@ def test_parse_and_validate_invalid_missing_kid_header():
     assert str(exception.value) == 'key_id cannot be empty.'
 
 
+def test_parse_and_validate_invalid_missing_sub():
+    rsa_key = rsa.generate_private_key(public_exponent=65537, key_size=2048)
+    jwt_bundle = JwtBundle(TrustDomain('test.orgcom'), {'kid1': rsa_key.public_key()})
+    rsakeypem, _ = get_keys_pems(rsa_key)
```
Comment (Andres-GC) [inline test/svid/jwtsvid/test_jwt_svid.py:237]: nit: `rsa_key_pem` looks clearer to me than `rsakeypem`.

An alternative to  `rsa_key_pem, _ = get_keys_pems(rsa_key)`is: `rsa_key_pem = get_keys_pems(rsa_key)[1]`

`get_keys_pems()` returns two values what means that is returning a tuple. That is why you can access what you need here using the index inside that returned tuple. I think this is it _pythonic_ way to do it.

**Review by Andres-GC — DISMISSED** [review]

File: src/pyspiffe/svid/jwt_svid.py (hunk 1 above)
Comment (glaucimar) [reply src/pyspiffe/svid/jwt_svid.py:135]: Thanks!

File: test/svid/jwtsvid/test_jwt_svid.py (hunk 2 above)
Comment (glaucimar) [reply test/svid/jwtsvid/test_jwt_svid.py:237]: I understand this... but I have seem both in Python, so, I am not so sure this is the _pythonic_ way... I will make the variable change name.

**Review by Andres-GC — APPROVED** [review]
Thanks Glaucimar

**Review by maxlambrecht — APPROVED** [review]

**Pull request merged**